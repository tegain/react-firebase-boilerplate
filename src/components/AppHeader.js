import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { startLogout } from "../actions/auth";

export const AppHeader = ({ startLogout }) => (
	<header className="header">
		<div className="header__container container">
			<h1 className="header__title">
				<Link to="/dashboard" className="header__link">
					React Firebase boilerplate
					<span>/dashboard</span>
				</Link>
			</h1>
			<button className="header__logout button button--transparent" onClick={startLogout}>Logout</button>
		</div>
	</header>
)

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
})


export default connect(undefined, mapDispatchToProps)(AppHeader)