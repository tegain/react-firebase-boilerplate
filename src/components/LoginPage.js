import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export const LoginPage = ({ startLogin }) => (
	<div className="box-layout">
		<div className="box-layout__box">
			<h1 className="box-layout__title">React Firebase boilerplate</h1>
			<p>Tag line for app</p>
			<button
				className="box-layout__btn button button--blue"
				onClick={startLogin}
			>
				Login with Google
			</button>
		</div>
	</div>
)

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)