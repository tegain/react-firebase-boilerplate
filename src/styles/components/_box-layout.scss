.box-layout {
	background: url('/images/bg.jpg') 50% 50% no-repeat;
	background-size: cover;
	height: 100vh;
	width: 100%;
	display: flex;
	justify-content: center;
	align-items: center;

	&__box {
		max-width: 35rem;
		border-radius: 2px;
		background: fade-out(#fff, .15);
		padding: $xl-size;
		text-align: center;
	}

	&__title {
		margin-top: 0;
		font-size: 2.6rem;
		line-height: 1.1;
	}
}