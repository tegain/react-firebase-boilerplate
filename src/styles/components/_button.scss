.button {
	padding: $m-size $l-size;
	border: none;
	border-radius: 2px;
	display: inline-block;
	color: #fff;
	text-decoration: none;
	transition: background .3s;

	&--blue {
		background: $blue;

		&:hover {
			background: darken($blue, 5%);
		}
	}

	&--transparent {
		background: none;
	}

	&--ghost {
		background-color: transparent;
		border: 1px solid #ccc;
		color: $dark-gray;

		&:hover {
			background: #f5f5f5;
		}
	}
}