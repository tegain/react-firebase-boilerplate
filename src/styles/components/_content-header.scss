.content-header {
	background: #f5f5f5;
	padding: $xl-size 0;

	&__heading {
		margin: 0;
		font-weight: 300;

		strong {
			font-weight: 700;
		}
	}

	&__actions {
		margin-top: $m-size;
	}
}