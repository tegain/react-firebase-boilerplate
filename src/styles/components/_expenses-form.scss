.expenses-form {
	width: 100%;

	.input-group__item,
	textarea {
		margin-right: 0;
		margin-bottom: $m-size;
	}

	.input-text {
		width: 100%;
	}
}