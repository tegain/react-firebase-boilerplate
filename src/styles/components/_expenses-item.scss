.expense-item {
	text-align: left;
	transition: background .25s;
	
	&:hover {
		background: #f7f7f7;
	}

	a {
		display: flex;
		flex-direction: column;
		justify-content: space-between;
		padding: $m-size;
		text-decoration: none;
		color: inherit;
		border-bottom: 1px solid #eee;

		@media screen and (min-width: $bp-medium) {
			flex-direction: row;
			align-items: center;
		}
	}

	&__content {
		flex: 3;
	}

	&__title {
		margin-top: 0;
		margin-bottom: .75rem;
		font-size: $m-size;
		word-break: break-all;
	}

	&__date {
		color: #666;
		font-size: 1.4rem;
	}

	&__note {
		position: relative;
		margin-top: $m-size;
		padding-top: $m-size;
		font-style: italic;
		color: #666;

		&:before {
			content: "";
			position: absolute;
			top: 0;
			left: 0;
			width: 2rem;
			height: 4px;
			background: #ccc;
		}
	}

	&__amount {
		flex: 1;
		margin-top: $l-size;
		font-weight: 700;
		font-size: $m-size;

		@media screen and (min-width: $bp-medium) {
			text-align: right;
			margin-top: 0;
		}
	}
}