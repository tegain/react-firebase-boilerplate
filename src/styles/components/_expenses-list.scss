.expenses-header {
	padding: $m-size;
	display: flex;
	justify-content: space-between;
	background: #f7f7f7;
	border: 1px solid #ddd;

	&__expense,
	&__amount {
		color: #666;
	}
}

.expenses-list {
		border: 1px solid #ddd;
		border-top: none;
		text-align: center;

	&__empty {
		padding: $xl-size $m-size;
		margin: 0;
	}
}