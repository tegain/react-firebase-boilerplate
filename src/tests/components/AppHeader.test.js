import React from 'react'
import { shallow } from 'enzyme'
import { AppHeader } from '../../components/AppHeader'

/**
 * Creats a snapshot of the component (logging the rendered component into a file),
 * then, when re-testing it, compares it to the snapshot version.
 * We can then choose to correct the mistakes, or update the snapshot version (pressing 'u')
 */
// test('should render AppHeader correctly using react-test-renderer/shallow', () => {
// 	const renderer = new ReactShallowRenderer
// 	renderer.render(<AppHeader />)
// 	expect(renderer.getRenderOutput()).toMatchSnapshot()
// 	console.log(renderer.getRenderOutput())
// })

test('should render <AppHeader /> correctly', () => {
	const wrapper = shallow(<AppHeader startLogout={() => {}} />)
	expect(wrapper).toMatchSnapshot()
})

test('should call startLogout on button click', () => {
	const startLogout = jest.fn()
	const wrapper = shallow(<AppHeader startLogout={startLogout} />)

	wrapper.find('button').simulate('click')
	expect(startLogout).toHaveBeenCalled()
})