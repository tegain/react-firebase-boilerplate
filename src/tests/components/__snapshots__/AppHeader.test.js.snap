// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <AppHeader /> correctly 1`] = `
<header
  className="header"
>
  <div
    className="header__container container"
  >
    <h1
      className="header__title"
    >
      <Link
        className="header__link"
        replace={false}
        to="/dashboard"
      >
        React Firebase boilerplate
        <span>
          /dashboard
        </span>
      </Link>
    </h1>
    <button
      className="header__logout button button--transparent"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
